### DISCUSSION TITLE
Clear instruction about how to change the Memmap temp directory

### DISCUSSION THREAD

**Discussion by sulli419** [flatironinstitute/CaImAn#1420] [Q&A]
I am trying to run CaImAn exclusively from a hard drive other than my OS (where python is installed).  At the very least, I need to change where the giant memmap temp files are saved.  For example, if I run on files from the D drive it keeps sticking the giant 32-bit temp memmap files in C.  Can someone please show me the exact code I would use to save the temp files to D instead?  

I think the answer might relate to this but it would help to get an example line of code that achieve this:
https://github.com/flatironinstitute/CaImAn/blob/main/caiman/paths.py

Thanks

**Comment by programerkotik** [comment]
Hi,

I can't answer your question, but I can relate to your experience. I attempted to amend the source code to specify the location, but it led me nowhere. There are numerous functions nested within each other, and you would need to pass the updated value through all of them if you want it as a parameter. Ultimately, I found it much easier to implement a copy-and-remove strategy at the end of each complete processing when running several files in a loop. However, this approach still requires enough space on drive C to save at least one file.

I hope you find a solution. There is also a related issue on the main page: https://github.com/flatironinstitute/CaImAn/issues/1419.

Best wishes

**Comment by steinkki** [comment]
Hello! 
on the related issue @programerkotik linked, I posted this question, and it was answered! Here is what I posted on the other discussion, which was verified to be the solution by others.

"
Through looking at other discussions, I was able to cobble together two possible solutions:

add the line os.environ['CAIMAN_DATA']='/path/to/new/data/directory' to the demo_pipeline.ipynb (particularly in the first code chunk, where there are similar lines of code towards the bottom).
-or-
add export CAIMAN_DATA=/path/to/new/data/directory to the activate.d file in the caiman env. (this solution is more permanent and you don't have to explicitly define the new path everytime you run the notebook, but it requires some digging to find. If you like this answer better, I can provide details on how I found the file)

"

hope this helps!
